CMakeLists.txt

```diff
@@ -208,26 +223,32 @@ set(C_CORESERVICES   "${DOSBOX_PLATFORM_MACOS}")
 set(C_ALSA "${DOSBOX_PLATFORM_LINUX}")
 
 # Windows
-set(NOMINMAX ON)
-
-# TODO Needs porting to `[[likely]]` and friends?
-set(C_HAS_BUILTIN_EXPECT OFF)
+if(DOSBOX_PLATFORM_WINDOWS)
+  # Prevent <windows.h> from clobbering std::min and std::max
+  set(NOMINMAX ON)
+  # Enable mathematical constants (such as M_PI) in Windows math.h header
+  set(_USE_MATH_DEFINES ON)
+endif()
 
-# TODO Check other functions
-check_symbol_exists(strnlen       "string.h"   HAVE_STRNLEN)
-check_symbol_exists(mprotect      "sys/mman.h" HAVE_MPROTECT)
-check_symbol_exists(mmap          "sys/mman.h" HAVE_MMAP)
-check_symbol_exists(MAP_JIT       "sys/mman.h" HAVE_MAP_JIT)
-check_symbol_exists(clock_gettime "time.h"     HAVE_CLOCK_GETTIME)
+check_symbol_exists(strnlen       "string.h"       HAVE_STRNLEN)
+check_symbol_exists(clock_gettime "time.h"         HAVE_CLOCK_GETTIME)
+check_symbol_exists(mprotect      "sys/mman.h"     HAVE_MPROTECT)
+check_symbol_exists(mmap          "sys/mman.h"     HAVE_MMAP)
+check_symbol_exists(MAP_JIT       "sys/mman.h"     HAVE_MAP_JIT)
+check_symbol_exists(setpriority   "sys/resource.h" HAVE_SETPRIORITY)
 
+check_symbol_exists(
+  pthread_setname_np "pthread.h" HAVE_PTHREAD_SETNAME_NP
+)
 check_symbol_exists(
   pthread_jit_write_protect_np "pthread.h" HAVE_PTHREAD_WRITE_PROTECT_NP
 )
 check_symbol_exists(
   sys_icache_invalidate "libkern/OSCacheControl.h" HAVE_SYS_ICACHE_INVALIDATE
 )
 
-set(_USE_MATH_DEFINES ON)
+# TODO: Detect PAGESIZE using contrib/pagesize/*_.pagesize.c
+
```

Comment:
I'm looking at the [commit notes](https://github.com/dosbox-staging/dosbox-staging/commit/7e20f6e401956a7a308f1b3462294d7ac9fa5db8) of pagesize.c and I'm scratching my head. The CPU page size is 4k on x86 and ARM. According to the comment, we'd get the wrong page size when cross-compiling for Alpha and Itanium processors... which we don't do, neither do we support running Staging on expensive server hardware, lol 

Then what is just use 4k on these platforms as well? What's the harm. Probably nothing as this seems to be some performance optimisation heuristics to align memoery block sizes with the page size or something.

I'd say we just kill this with Meson.

Reply:
4k is standard on common platforms. The reason is unclear why we need to be super precise about this. It's an optimisation thing that might yield zero benefit, so I'd say kill it.

Reply:
It's great you looked into it a bit more deeply @FeralChild64 🎉 Yeah these optimisation attempts need to be backed up by real world profiler metrics, otherwise it becomes just "cargo culting" to keep them alive forever and they just add complexity.

Hardcoding to 16k on macOS ARM is the appropiate solution, so good call. 👍🏻